api/views/workflow.py

```diff
@@ -228,3 +226,23 @@ def post(self, request, *args, **kwargs):
             return Response({'task_id': async_result.id}, status=status.HTTP_202_ACCEPTED)
         else:
             return Response({'errors': serializer.errors, 'data': prelim_data}, status=status.HTTP_400_BAD_REQUEST)
+
+    def _get_source_config(self, user):
+        config_label = '{}.v1_push'.format(user.username)
+        try:
+            return SourceConfig.objects.get(label=config_label)
```

Comment:
I think we should just return a 400 or 403 if the config is not found? Use of older APIs should be explicitly allowed by us.